shopinvader/views/shopinvader_variant_view.xml

```diff
@@ -9,7 +9,7 @@
         <field name="model">shopinvader.variant</field>
         <field name="arch" type="xml">
             <search string="Products">
-                <field name="name" string="Product"/>
+                <field name="name" string="Product" filter_domain="['|','|','|',('default_code','ilike',self),('name','ilike',self),('barcode','ilike',self),('shopinvader_name', 'ilike', self)]"/>
```

Comment:
/s/filter_domain/domain

IMO it's better to specify the domain into the field definition to ease the override (as a @api.model method)